sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -81,6 +89,109 @@ def __init__(
+        ):
+            raise ValueError("One of the required parameters is None")
+
+        if num_procs_per_worker > resources_per_worker["gpu"]:
+            raise ValueError("Insufficient gpu resources allocated to the container.")
+
+        if isinstance(model_params, HuggingFaceModelParams):
+            mp = "hf"
+
+        if isinstance(dataset_params, S3DatasetParams):
+            dp = "s3"
+
+        # create init container spec
+        init_container_spec = utils.get_container_spec(
+            name=constants.JOB_PARAMETERS[constants.PYTORCHJOB_KIND]["init_container"],
+            image=constants.JOB_PARAMETERS[constants.PYTORCHJOB_KIND][
+                "init_container_image"
+            ],
+            args=[
+                "--model_provider",
+                mp,
+                "--model_provider_args",
+                json.dumps(model_params.__dict__),
+                "--dataset_provider",
+                dp,
+                "--dataset_provider_args",
+                json.dumps(dataset_params.__dict__),
+            ],
+            volume_mounts=models.V1VolumeMount(
+                name="model_dataset_store", mount_path="/workspace"
+            ),
+        )
+
+        # create app container spec
+        container_spec = utils.get_container_spec(
+            name=constants.JOB_PARAMETERS[constants.PYTORCHJOB_KIND]["container"],
+            image=constants.JOB_PARAMETERS[constants.PYTORCHJOB_KIND][
+                "train_container_image"
+            ],
+            args=["--parameters", json.dumps(parameters.__dict__)],
+            volume_mounts=models.V1VolumeMount(
+                name=pvc["name"], mount_path="/workspace"
+            ),
+            resources=resources_per_worker,
+        )
+
+        # create worker pod spec
+        worker_pod_template_spec = utils.get_pod_template_spec(
+            job_kind=constants.PYTORCHJOB_KIND,
+            containers_spec=[container_spec],
+            volumes_spec=[
+                models.V1Volume(
+                    name=pvc["name"], persistent_volume_claim=pvc["claimName"]
+                )
+            ],
+        )
+
+        # create master pod spec
+        master_pod_template_spec = utils.get_pod_template_spec(
+            job_kind=constants.PYTORCHJOB_KIND,
+            containers_spec=[init_container_spec, container_spec],
+            volumes_spec=[
+                models.V1Volume(
+                    name=pvc["name"], persistent_volume_claim=pvc["claimName"]
+                )
+            ],
+        )
+
+        job = utils.get_pytorchjob_template(
+            name=name,
+            namespace=namespace,
+            master_pod_template_spec=master_pod_template_spec,
+            worker_pod_template_spec=worker_pod_template_spec,
+            num_worker_replicas=num_workers,
+            num_procs_per_worker=num_procs_per_worker,
+            elastic_policy=models.KubeflowOrgV1ElasticPolicy(rdzv_backend="c10d"),
```

Comment:
Without elastic policy currently we can't user torchrun command ?
@kuizhiqing @johnugeorge  ?

Reply:
I tried we can use, but since elastic spec already exists, I chose to use it.

Reply:
@andreyvelich Yes, we can use `torchrun` without elasticPolicy if we use the version after PR https://github.com/kubeflow/training-operator/pull/1840.

Reply:
@deepanker13 Do we need to run `torchrun` with elastic policy in the first version for `train` API ?
Maybe we can introduce that feature in the future ?

Reply:
Isn't it better to use elastic policy by default? What problem do you see?

Reply:
Elastic policy makes sense when we set min and max replicas, what is the reason to just set the c10d backend ?

Reply:
@andreyvelich I think we can min and max replicas option as optional parameter and use elastic policy spec to launch torchrun.

Reply:
@deepanker13 Can we introduce those changes in the following PRs when we have initial version of `train` API working ?

Reply:
@andreyvelich without elastic I have to pass these args like this in the spec like this:-
```
              args: ["torchrun", "--nproc_per_node=1","--rdzv_backend=c10d","--rdzv_id=test","--rdzv_endpoint=torchrun-test-worker-0:29400","/workspace/exp/training.py"]
```

and with elastic only:-
```
              args: ["torchrun","/workspace/exp/training.py"]
```
Isn't the second one better?

Reply:
@deepanker13 Why do you need to set it ?
Check this PR: https://github.com/kubeflow/training-operator/pull/1965/files
@kuizhiqing Submitted example where you just need to set the nprocPerNode: "2" API and then you can use the `torchrun train.py` command.